### PR TITLE
Kubernetes docs

### PR DESCRIPTION
**Pull request by franciscolopezsancho**
### What changes were proposed in this pull request?
adding annotation and ports documentation while taking kubernates configuration out of the current page

### Why are the changes needed?
to explain the new capabilities and ease the read

### Does this PR introduce any user-facing change?
yes the docs

### How was this patch tested?
running the local-preview

### CODE REVIEW

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+:page-partial:
+
+include::ROOT:partial$include.adoc[]
+
+== Available Kubernetes Settings 
+
+In this section we are covering the different Kubernetes parameters that are configurable. 
+
+To set the configuration on a container related to Kubernetes is necessary, first of all, to add that configuration under `cloudflow.streamlets.[streamlet-name].kubernetes`.
+On the followint samples we show configurations under a specific streamlet, `my-streamlet`, that in its complete path is `cloudflow.streamlets.my-streamlet`. It is also possible, as explained in xref:develop:cloudflow-configuration.adoc#_configuring_a_runtime_using_the_runtime_scope[Configuring a Runtime using the runtime Scope], to apply any configuration shown in here to any `runtimes`. The complete path for a runtime is, for instance, `cloudflow.runtimes.akka`.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:11]:
```suggestion
On the following samples we show configurations under a specific streamlet, `my-streamlet`, that in its complete path is `cloudflow.streamlets.my-streamlet`. It is also possible, as explained in xref:develop:cloudflow-configuration.adoc#_configuring_a_runtime_using_the_runtime_scope[Configuring a Runtime using the runtime Scope], to apply any configuration shown in here to any `runtimes`. The complete path for a runtime is, for instance, `cloudflow.runtimes.akka`.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+=== Resources and Environment Variables
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:13]: Please create separate sections for configuring resources and configuring environment variables. 
"Configure container resource requests and limits" could be a title for one, "Configure container environment variables" for the other.

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 3]
```diff
@@ -0,0 +1,363 @@
+
+:page-partial:
+
+include::ROOT:partial$include.adoc[]
+
+== Available Kubernetes Settings 
+
+In this section we are covering the different Kubernetes parameters that are configurable. 
+
+To set the configuration on a container related to Kubernetes is necessary, first of all, to add that configuration under `cloudflow.streamlets.[streamlet-name].kubernetes`.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:10]:
```suggestion
Several Kubernetes settings for a streamlet can be set under `cloudflow.streamlets.[streamlet-name].kubernetes`.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc (hunk 1 above)
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:11]:
```suggestion
In the following examples we show configurations for a specific streamlet named `my-streamlet`, under `cloudflow.streamlets.my-streamlet`. It is also possible, as explained in xref:develop:cloudflow-configuration.adoc#_configuring_a_runtime_using_the_runtime_scope[Configuring a Runtime using the runtime Scope], to apply any configuration shown here to a specific `runtime`. For example, Kubernetes settings for all akka streamlets can be set under`cloudflow.runtimes.akka.kubernetes`.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+On the followint samples we show configurations under a specific streamlet, `my-streamlet`, that in its complete path is `cloudflow.streamlets.my-streamlet`. It is also possible, as explained in xref:develop:cloudflow-configuration.adoc#_configuring_a_runtime_using_the_runtime_scope[Configuring a Runtime using the runtime Scope], to apply any configuration shown in here to any `runtimes`. The complete path for a runtime is, for instance, `cloudflow.runtimes.akka`.
+
+=== Resources and Environment Variables
+
+Container resource requirements and environment variables can be set in the `cloudflow.streamlets.[streamlet-name].kubernetes.pods.pod.container.container` section.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:15]:
```suggestion
Container resource requirements and environment variables can be set in the `cloudflow.streamlets.[streamlet-name].kubernetes.pods.pod.containers.container` section.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+}
+----
+
+The above example shows that `500mcpu` is requested, as well as `512Mi` of memory. 
+The pod is limited to use `4Gi` of memory. It also shows how to set the environment variable `JAVA_OPTS=XX:MaxRAMPercentage=40.0` and `FOO=BAR`  
+
+
+===  Labels
+
+To set label into a container we need to add the `label` setting as the example below.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:51]:
```suggestion
To set container labels, add the `labels` section shown in the example below:
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+  kubernetes.pods.pod {
+    labels {
+      key1 = value1
+      key2 = value2
+    }
+  }
+}
+----
+
+The example above shows is adding the labels `key1 = value1` and `key2 = value2` to the pod that contains the streamlet `my-streamlet`.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:65]:
```suggestion
The example above shows how to add the labels `key1 = value1` and `key2 = value2` to the pod for streamlet `my-streamlet`. 
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 7]
```diff
@@ -0,0 +1,363 @@
...
+    }
+  }
+}
+----
+
+The example above shows is adding the labels `key1 = value1` and `key2 = value2` to the pod that contains the streamlet `my-streamlet`. 
+
+===  Annotations
+
+Similar to labels we can add annotations to any container. This snippet shows how
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:69]:
```suggestion
Similar to labels we can add annotations to any streamlet pod. The snippet below shows how:
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 8]
```diff
@@ -0,0 +1,363 @@
...
+            key1 = annotation1
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+This configuration adds the anotation `key: annotation1` to the container where `my-akka-streamlet` is deployed.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:88]:
```suggestion
This configuration adds the annotation `key: annotation1` to the pod for streamlet `my-akka-streamlet`.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 9]
```diff
@@ -0,0 +1,363 @@
...
+
+Annotations, same as labels, are defined by a key/value and also can be defined specifically for a streamlet or generically to a runtime, like cloudflow.runtime.akka.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:90]:
```suggestion
Annotations, same as labels, are defined by a key/value and also can be defined specifically for a streamlet or for all streamlets of a runtime, under `cloudflow.runtimes.akka.kubernetes.pods.pod.annotations`. 
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 10]
```diff
@@ -0,0 +1,363 @@
...
+}
+----
+
+This configuration adds the anotation `key: annotation1` to the container where `my-akka-streamlet` is deployed.
+
+Annotations, same as labels, are defined by a key/value and also can be defined specifically for a streamlet or generically to a runtime, like cloudflow.runtime.akka. 
+
+===  Ports
+
+To open ports in container we need to add at least the port number. This snippet shows how
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:94]:
```suggestion
To open ports in containers we need to at least add a port number. The snippet below shows how:
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 11]
```diff
@@ -0,0 +1,363 @@
...
+    }
+  }
+}
+----
+
+This configuration adds the anotation `key: annotation1` to the container where `my-akka-streamlet` is deployed.
+
+Annotations, same as labels, are defined by a key/value and also can be defined specifically for a streamlet or generically to a runtime, like cloudflow.runtime.akka. 
+
+===  Ports
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:92]:
```suggestion
===  Container Ports
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 12]
```diff
@@ -0,0 +1,363 @@
...
+          ]
+        }
+      }
+    }
+  }
+}
+----
+
+It is possible, as we showed above, to also set host-ip, protocol, host-port and a name. 
+This configuration opens in the container where `my-akka-streamlet` is deployed two ports. One in 9001 as TCP and the other in 9002 as UDP with `Host Ports:9999/UDP`, `name: user-datagram` and `hostIp: 10.132.0.123`
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:115]:
```suggestion
This configuration adds two container ports to every container where `my-akka-streamlet` is deployed. One in 9001 as TCP and the other in 9002 as UDP with `Host Ports:9999/UDP`, `name: user-datagram` and `hostIp: 10.132.0.123`
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 13]
```diff
@@ -0,0 +1,363 @@
...
+
+For more info on kubernetes API https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#containerport-v1-core[container port]
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:117]:
```suggestion
For more info see container ports in the Kubernetes API https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#containerport-v1-core[container port]
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 14]
```diff
@@ -0,0 +1,363 @@
...
+
+Note:
+Only Akka Streamlets can open ports
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:120]:
```suggestion
It is only possible to add container ports to Akka Streamlets, the Flink and Spark operator do not support this.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 15]
```diff
@@ -0,0 +1,363 @@
...
+
+For more info on kubernetes API https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#containerport-v1-core[container port]
+
+Note:
+Only Akka Streamlets can open ports 
+
+=== Volumes and Volume-Mounts
+==== **Mounting Secrets**
+
+It possible to add `volumes`, that contain a secret, and mount them into the container with `volume-mounts`. Bear in mind that each `kubernetes.pods.pod.containers.container.volume-mounts` name must match a `kubernetes.pods.pod.volumes` name in the config.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:125]:
```suggestion
It possible to add a `volume` that contains a secret, and mount it into the container with `volume-mounts`. Bear in mind that each `kubernetes.pods.pod.containers.container.volume-mounts` name must match a `kubernetes.pods.pod.volumes` name in the config. 
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 16]
```diff
@@ -0,0 +1,363 @@
...
+| `volumes` | `kubernetes.pods.pod.volumes.[name].pvc.name = [pvc-name]`
+| `volumes` | `kubernetes.pods.pod.volumes.[name].pvc.read-only = [true or false]`
+| `volume-mounts` | `pod.containers.container.volume-mounts.[name].mount-path = [some-path]`
+| `volume-mounts` | `pod.containers.container.volume-mounts.[name].read-only = [true or false]`
+|===
+
+
+=== Pod differences among Akka, Flink and Spark Streamlets.
+
+Akka runtimes only deploy one pod per streamlet while Flink and Sparks deploy two. Spark creates a Driver pod and a Executor pod, while Flink creates a JobManager and a TaskManager.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:231]:
```suggestion
Akka runtimes only deploy one type of pod per streamlet. Spark creates Driver and Executor pods, while Flink creates JobManager and TaskManager pods.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 17]
```diff
@@ -0,0 +1,363 @@
...
+
+When we are setting Kubernetes configuration via `kubernetes.pods.pod` or `kubernetes.pods.pod.containers.container`, it applies the same settings to all pods. Whether is one or two.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:233]:
```suggestion
When we are setting Kubernetes configuration via `kubernetes.pods.pod` or `kubernetes.pods.pod.containers.container`, it applies the same settings to all types of pod for the runtime used.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 18]
```diff
@@ -0,0 +1,363 @@
...
+Akka runtimes only deploy one pod per streamlet while Flink and Sparks deploy two. Spark creates a Driver pod and a Executor pod, while Flink creates a JobManager and a TaskManager.
+
+When we are setting Kubernetes configuration via `kubernetes.pods.pod` or `kubernetes.pods.pod.containers.container`, it applies the same settings to all pods. Whether is one or two. 
+
+In the case of Spark we can be more specific by using the following paths:
+  
+- For the Driver pods: `kubernetes.pods.driver` and `kubernetes.pods.driver.containers.container`     
+- For the Executor pods: `kubernetes.pods.executor` and `kubernetes.pods.executor.containers.container`   
+
+Using the path to the `pod` (driver or executor) or the container it will solely depend on the parameter we are setting.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:240]:
```suggestion
Using the path to the `pod` (driver or executor) settings will only applied to that specific type of pod. 
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 19]
```diff
@@ -0,0 +1,363 @@
...
+When we are setting Kubernetes configuration via `kubernetes.pods.pod` or `kubernetes.pods.pod.containers.container`, it applies the same settings to all pods. Whether is one or two. 
+
+In the case of Spark we can be more specific by using the following paths:
+  
+- For the Driver pods: `kubernetes.pods.driver` and `kubernetes.pods.driver.containers.container`     
+- For the Executor pods: `kubernetes.pods.executor` and `kubernetes.pods.executor.containers.container`   
+
+Using the path to the `pod` (driver or executor) or the container it will solely depend on the parameter we are setting. 
+
+Flink configuration works similarly to Spark's as we can set the following paths:
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:242]:
```suggestion
Flink configuration works similarly as we can set the following paths:
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 20]
```diff
@@ -0,0 +1,363 @@
...
+    } 
+  }
+}
+---- 
+
+This configuration will mount `mysecret1` into any `driver` pod on path `/mnt/folderA/mysecret1` as read-only and into any `executor. It will mount `myclaim` into any `driver` pod on path `/mnt/folderB` as writable and `mysecret2` will be mounted into any `executor` pod on path `/mnt/folderC/mysecret2`. Inside folders `A` and `C`, you'll find as many files as values are under the https://kubernetes.io/docs/concepts/configuration/secret/#decoding-a-secret[`data`] tag in the Kubernetes Secret mounted. While the mounted `myclaim1` into the `driver` pods will allow to share the contents of folder `/mnt/folderB` among all of them. 
+
+## Exceptions
+
+Is important to bear in mind that flink configuration will not be able to set specific labels per type of pod (`task-manager` and `job-manager` pods). Only `cloudflow.runtimes.flink.kubernetes.pods.pod.labels` is allowed, cloudflow.runtimes.flink.kubernetes.pods.[pod type].labels is not. It has the same limitation when defining `volumes` and `volume-mounts`. Is only allowed `cloudflow.runtimes.flink.kubernetes.pods.pod.volumes` and `cloudflow.runtimes.flink.kubernetes.pods.pod.containers.container.volume-mounts`.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:311]:
```suggestion
Is important to bear in mind that Flink configuration will not be able to set specific labels per type of pod (`task-manager` and `job-manager` pods). Only `cloudflow.runtimes.flink.kubernetes.pods.pod.labels` is allowed, cloudflow.runtimes.flink.kubernetes.pods.[pod type].labels is not. It has the same limitation when defining `volumes` and `volume-mounts`. Is only allowed `cloudflow.runtimes.flink.kubernetes.pods.pod.volumes` and `cloudflow.runtimes.flink.kubernetes.pods.pod.containers.container.volume-mounts`.
```

**Review by RayRoestenburg — COMMENTED** [review]
Please accept my suggestions, looks good otherwise.

**Comment by RayRoestenburg** [comment]
Also please create separate sections for configuring resources and configuring environment variables @franciscolopezsancho better for searching and simpler to read one thing at a time.

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc (hunk 3 above)
Comment (franciscolopezsancho) [reply docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:10]: nice

**Comment by franciscolopezsancho** [comment]
I changed the overall structure of all the docs regarding configuration. Wasn't that much after all

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration.adoc [hunk 21]
```diff
@@ -28,74 +19,25 @@ We have two main top-level scopes, one for streamlets and one for runtimes, as w
...
-<1> `config-parameters` section
-<2> `config` section
-<3> `kubernetes` section
-
-You need to specify at least one of the sections shown above.
+Using this model, you can configure the following settings:
 
-==== Environment variables
+    - Runtime settings (for Akka, Spark, Flink, or any user-provided one)
+    - Kubernetes container resource requirements
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration.adoc:29]:
```suggestion
    - Kubernetes pod and container configuration
```

File: docs/docs-source/docs/modules/ROOT/nav.adoc [hunk 22]
```diff
@@ -35,7 +35,10 @@ include::partial$include.adoc[]
 *** xref:develop:use-flink-streamlets.adoc[Using Flink Streamlets]
 **** xref:develop:build-flink-streamlets.adoc[Building a Flink Streamlet]
 **** xref:develop:test-flink-streamlet.adoc[Testing a Flink Streamlet]
-*** xref:develop:streamlet-configuration.adoc[Configuring Streamlets]
+*** xref:develop:cloudflow-configuration.adoc[The Configuration Model]
```
Comment (RayRoestenburg) [inline docs/docs-source/docs/modules/ROOT/nav.adoc:38]:
```suggestion
*** xref:develop:cloudflow-configuration.adoc[Configuring Streamlets]
```

Comment (RayRoestenburg) [reply docs/docs-source/docs/modules/ROOT/nav.adoc:38]: On the same level as Developing Streamlets, I think makes sense.

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration.adoc [hunk 23]
```diff
@@ -4,22 +4,13 @@ include::ROOT:partial$include.adoc[]
 
 Cloudflow offers a flexible configuration model that lets you configure every aspect of the application deployment.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration.adoc:5]:
```suggestion
Cloudflow offers a flexible configuration model that lets you configure every aspect of application deployment.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 24]
```diff
@@ -0,0 +1,373 @@
+
+:page-partial:
+
+include::ROOT:partial$include.adoc[]
+
+== Available Kubernetes Settings 
+
+In this section we are covering the different Kubernetes parameters that are configurable.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:8]:
```suggestion
In this section we are covering the different Kubernetes settings that are configurable. 
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 25]
```diff
@@ -0,0 +1,373 @@
...
+:page-partial:
+
+include::ROOT:partial$include.adoc[]
+
+== Available Kubernetes Settings 
+
+In this section we are covering the different Kubernetes parameters that are configurable. 
+
+Several Kubernetes settings for a streamlet can be set under `cloudflow.streamlets.[streamlet-name].kubernetes`.
+On the following samples we show configurations under a specific streamlet, `my-streamlet`, that in its complete path is `cloudflow.streamlets.my-streamlet`. It is also possible, as explained in xref:develop:cloudflow-configuration.adoc#_configuring_a_runtime_using_the_runtime_scope[Configuring a Runtime using the runtime Scope], to apply any configuration shown in here to any `runtimes`. The complete path for a runtime is, for instance, `cloudflow.runtimes.akka`.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:11]:
```suggestion
On the following samples we show configurations under a specific streamlet, `my-streamlet`, its complete path is `cloudflow.streamlets.my-streamlet`. It is also possible, as explained in xref:develop:cloudflow-configuration.adoc#_configuring_a_runtime_using_the_runtime_scope[Configuring a Runtime using the runtime Scope], to apply any configuration shown in here to any `runtimes`. The complete path for a runtime is, for instance, `cloudflow.runtimes.akka`.
```

**Review by RayRoestenburg — COMMENTED** [review]
Some minor changes to be made, good improvements!

File: docs/docs-source/docs/modules/ROOT/nav.adoc (hunk 22 above)
Comment (franciscolopezsancho) [reply docs/docs-source/docs/modules/ROOT/nav.adoc:38]: It makes sense to me too. Question; if we change to that title then we have to pages with the same title
** xref:develop:cloudflow-configuration.adoc[Configuring Streamlets]
**** xref:develop:streamlet-configuration.adoc[Configuring Streamlets]

Comment (RayRoestenburg) [reply docs/docs-source/docs/modules/ROOT/nav.adoc:38]: streamlet-configuration is not the right title for that anyway, it is the specific feature of Configuration Parameters.
Please rename `**** xref:develop:streamlet-configuration.adoc[Configuring Streamlets]` to `**** xref:develop:streamlet-configuration.adoc[Streamlet Config Parameters]`, probably best to also rename the file to streamlet-config-parameters.adoc @franciscolopezsancho .

**Comment by franciscolopezsancho** [comment]
back at you :) @RayRoestenburg

File: docs/docs-source/docs/modules/ROOT/nav.adoc [hunk 26]
```diff
@@ -35,7 +35,10 @@ include::partial$include.adoc[]
 *** xref:develop:use-flink-streamlets.adoc[Using Flink Streamlets]
 **** xref:develop:build-flink-streamlets.adoc[Building a Flink Streamlet]
 **** xref:develop:test-flink-streamlet.adoc[Testing a Flink Streamlet]
-*** xref:develop:streamlet-configuration.adoc[Configuring Streamlets]
+** xref:develop:cloudflow-configuration.adoc[Configuring Streamlets]
+*** xref:develop:cloudflow-configuration-kubernetes.adoc[Kubernetes Configuration]
+*** xref:develop:cloudflow-configuration-scopes.adoc[]
+*** xref:develop:streamlet-configuration.adoc[Streamlet Config Params]
```
Comment (RayRoestenburg) [inline docs/docs-source/docs/modules/ROOT/nav.adoc:41]:
```suggestion
*** xref:develop:streamlet-configuration.adoc[Streamlet Configuration Parameters]
```

File: docs/docs-source/docs/modules/develop/pages/streamlet-configuration.adoc [hunk 27]
```diff
@@ -1,4 +1,4 @@
-= Streamlet Configuration
+= Streamlet Config Params
```
Comment (RayRoestenburg) [inline docs/docs-source/docs/modules/develop/pages/streamlet-configuration.adoc:1]:
```suggestion
= Streamlet Configuration Parameters 
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 28]
```diff
@@ -0,0 +1,373 @@
+
+:page-partial:
+
+include::ROOT:partial$include.adoc[]
+
+== Available Kubernetes Settings 
+
+In this section we are covering the different Kubernetes settings that are configurable.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:8]:
```suggestion
In this section we are covering the Kubernetes settings that you can configure in Cloudflow. 
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 29]
```diff
@@ -0,0 +1,373 @@
...
+:page-partial:
+
+include::ROOT:partial$include.adoc[]
+
+== Available Kubernetes Settings 
+
+In this section we are covering the different Kubernetes settings that are configurable. 
+
+Several Kubernetes settings for a streamlet can be set under `cloudflow.streamlets.[streamlet-name].kubernetes`.
+On the following samples we show configurations under a specific streamlet, `my-streamlet`, its complete path is `cloudflow.streamlets.my-streamlet`. It is also possible, as explained in xref:develop:cloudflow-configuration.adoc#_configuring_a_runtime_using_the_runtime_scope[Configuring a Runtime using the runtime Scope], to apply any configuration shown in here to any `runtimes`. The complete path for a runtime is, for instance, `cloudflow.runtimes.akka`.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:11]:
```suggestion
In the following examples we show configurations under a specific streamlet, `my-streamlet`, its complete path is `cloudflow.streamlets.my-streamlet`. It is also possible, as explained in xref:develop:cloudflow-configuration.adoc#_configuring_a_runtime_using_the_runtime_scope[Configuring a Runtime using the runtime Scope], to apply any configuration shown in here to any `runtimes`. The complete path for a runtime is, for instance, `cloudflow.runtimes.akka`.
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 30]
```diff
@@ -0,0 +1,373 @@
...
+
+=== Resources
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:13]:
```suggestion
=== Resource Requirements
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 31]
```diff
@@ -0,0 +1,373 @@
...
+        limits {
+          memory = "4Gi"
+        }
+      }
+    }
+  }
+}
+----
+
+The above example shows that `500mcpu` is requested, as well as `512Mi` of memory while being limited to use `4Gi` of memory.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:37]:
```suggestion
The above example shows that `500mcpu` and `512Mi` of memory is requested, while the memory limit is set to `4Gi`.   
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 32]
```diff
@@ -0,0 +1,373 @@
...
+              name = "FOO"
+              value = "BAR"
+            }
+          ]
+    }
+  }
+}
+----
+
+The above example shows how to set the environment variable `JAVA_OPTS=XX:MaxRAMPercentage=40.0` and `FOO=BAR`
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:62]:
```suggestion
The above example shows how to set the environment variables `JAVA_OPTS=XX:MaxRAMPercentage=40.0` and `FOO=BAR` 
```

File: docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc [hunk 33]
```diff
@@ -0,0 +1,373 @@
...
+        }
+      }
+    }
+  }
+}
+----
+
+This configuration adds the annotation `key: annotation1` to the pod for streamlet `my-akka-streamlet`.
+
+Annotations, same as labels, are defined by a key/value and also can be defined specifically for a streamlet or for all streamlets of a runtime, under `cloudflow.runtimes.akka.kubernetes.pods.pod.annotations`.
```
Comment (RayRoestenburg) [inline docs/shared-content-source/docs/modules/develop/pages/cloudflow-configuration-kubernetes.adoc:105]:
```suggestion
Annotations, similar to labels, are defined by a key/value and also can be defined specifically for a streamlet or for all streamlets of a runtime, under `cloudflow.runtimes.akka.kubernetes.pods.pod.annotations`. 
```

**Review by RayRoestenburg — COMMENTED** [review]
Some final comments!

**Review by RayRoestenburg — APPROVED** [review]

**Pull request merged**